### ISSUE TITLE
An error occurred when creating project from project template

### ISSUE THREAD

**Issue by kyungmi** [webida/webida-server#152]
```
2015-11-20 12:55:47-337 - info: src =  /webida/fs/xkADkKcOW/basic/Empty/project
2015-11-20 12:55:47-338 - info: dst =  /webida/fs/lJMveNmMae/abc/aaaaaaaa
2015-11-20 12:55:47-338 - info: NEW FILE COUNT =  2
2015-11-20 12:55:47-340 - error: failed to copy link: Error: SEC:failed to get file id
    at copyFileLink (/home/webida/webida-server/webida.net/server/fs/lib/flinkmap.js:493:16)
    at /home/webida/webida-server/webida.net/server/fs/lib/flinkmap.js:340:33
    at /home/webida/webida-server/webida.net/server/fs/lib/flinkmap.js:110:20
    at /home/webida/webida-server/webida.net/server/fs/lib/flinkmap.js:103:16
    at Query._callback (/home/webida/webida-server/webida.net/server/node_modules/data-mapper/lib/dao.js:93:13)
    at Query.Sequence.end (/home/webida/webida-server/webida.net/server/node_modules/data-mapper/node_modules/mysql/lib/protocol/sequences/Sequence.js:96:24)
    at Query._handleFinalResultPacket (/home/webida/webida-server/webida.net/server/node_modules/data-mapper/node_modules/mysql/lib/protocol/sequences/Query.js:144:8)
    at Query.EofPacket (/home/webida/webida-server/webida.net/server/node_modules/data-mapper/node_modules/mysql/lib/protocol/sequences/Query.js:128:8)
    at Protocol._parsePacket (/home/webida/webida-server/webida.net/server/node_modules/data-mapper/node_modules/mysql/lib/protocol/Protocol.js:271:23)
    at Parser.write (/home/webida/webida-server/webida.net/server/node_modules/data-mapper/node_modules/mysql/lib/protocol/Parser.js:77:12)
    at Protocol.write (/home/webida/webida-server/webida.net/server/node_modules/data-mapper/node_modules/mysql/lib/protocol/Protocol.js:39:16)
    at Socket.<anonymous> (/home/webida/webida-server/webida.net/server/node_modules/data-mapper/node_modules/mysql/lib/Connection.js:92:28)
    at emitOne (events.js:77:13)
    at Socket.emit (events.js:169:7)
    at readableAddChunk (_stream_readable.js:146:16)
    at Socket.Readable.push (_stream_readable.js:110:10)
2015-11-20 12:55:47-342 - error: failed to copy link: Error: SEC:failed to get file id
    at copyFileLink (/home/webida/webida-server/webida.net/server/fs/lib/flinkmap.js:493:16)
    at /home/webida/webida-server/webida.net/server/fs/lib/flinkmap.js:340:33
    at /home/webida/webida-server/webida.net/server/fs/lib/flinkmap.js:110:20
    at /home/webida/webida-server/webida.net/server/fs/lib/flinkmap.js:103:16
    at Query._callback (/home/webida/webida-server/webida.net/server/node_modules/data-mapper/lib/dao.js:93:13)
    at Query.Sequence.end (/home/webida/webida-server/webida.net/server/node_modules/data-mapper/node_modules/mysql/lib/protocol/sequences/Sequence.js:96:24)
    at Query._handleFinalResultPacket (/home/webida/webida-server/webida.net/server/node_modules/data-mapper/node_modules/mysql/lib/protocol/sequences/Query.js:144:8)
    at Query.EofPacket (/home/webida/webida-server/webida.net/server/node_modules/data-mapper/node_modules/mysql/lib/protocol/sequences/Query.js:128:8)
    at Protocol._parsePacket (/home/webida/webida-server/webida.net/server/node_modules/data-mapper/node_modules/mysql/lib/protocol/Protocol.js:271:23)
    at Parser.write (/home/webida/webida-server/webida.net/server/node_modules/data-mapper/node_modules/mysql/lib/protocol/Parser.js:77:12)
    at Protocol.write (/home/webida/webida-server/webida.net/server/node_modules/data-mapper/node_modules/mysql/lib/protocol/Protocol.js:39:16)
    at Socket.<anonymous> (/home/webida/webida-server/webida.net/server/node_modules/data-mapper/node_modules/mysql/lib/Connection.js:92:28)
    at emitOne (events.js:77:13)
    at Socket.emit (events.js:169:7)
    at readableAddChunk (_stream_readable.js:146:16)
    at Socket.Readable.push (_stream_readable.js:110:10)
2015-11-20 12:55:47-342 - info: ---------SEC: ------ recursive copy work for link is done

```

**Comment by gseok** [comment]
This issue connected on 'server and client' dependancy issue.

---------------------------------------------------------------------------------
error: failed to copy link
--> I Think.. This is ACL problem..

src =  /webida/fs/xkADkKcOW/basic/Empty/project
--> /fs/xkADkKcOW  is filesystem.
--> /fs/xkADkKcOW  file system contains all the templates.
--> Another file system get template (in /fs/xkADkKcOW) when project wizard's create operation

Actually, each file system is protected by an ACL.

---------------------------------------------------------------------------------

workaround

1. The server does not have a resource of templates.
2. The client does have templates. (already have templates in webida-client/apps/ide/ProjectWizard-templates)
3. The client does have all templates meta-file. (e.g. template-meta.json)
4. When create project, client send request URL(e.g. <myFsID>/getTemplate/templatePath, using param for copyed)
 - Then server copy to (apps/ide/ProjectWizard-templates  --> target).
 - or file send?


These workaround must require discussion.

-----

**Comment by gotchazipc** [comment]
Just fix ACL problem in 1.7.0 for now. 

We may have to provide a better way to import client-side resources to users workspace, withtout using shared-webida-fs. 

1) server may can provide some 'import' APIs to read external resources into webida fs. It looks clean and simple, but I beleive that such an approch will not be the right answer. What will happen if client uses 'local storage' as a workspace? What about Dropbox or GoogleDrive? 

2) client should read template resources by itself and upload the template file to server. To make job easier, any template for a project should be compressed into a zip file, and client should upload the file and extract it on webida fs.

**Comment by gotchazipc** [comment]
separating client-side-resources from server, is not fully implemented in webida 1.x. So, the ultimate solution for this issue should be a part of webida 2.x (both of server and client) and  my idea requires more works on client-side. Let's talk about this issue with client guys soon.
